lib/Doctrine/ORM/Mapping/MetadataCollection.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+
+use Doctrine\ORM\Utility\StaticClassNameConverter;
+
+class MetadataCollection
+{
+
+    /**
+     * @var ClassMetadata[]
+     */
+    private $metadata;
+
+    private function __construct(array $metadata)
+    {
+        $metadataByClassName = array_combine(
+            array_map(function (ClassMetadata $metadata) { return $metadata->getClassName(); }, $metadata),
+            $metadata
+        );
+        $this->metadata = $metadataByClassName;
+    }
+
+    public function get($name)
+    {
+        $name = StaticClassNameConverter::getRealClass($name);
+        if(!isset($this->metadata[$name])){
+            throw new \Exception('No metadata found for ' . $name);
+        }
+        return $this->metadata[$name];
+    }
+
+    public static function fromClassMetadatas(ClassMetadata $firstClass, ClassMetadata ...$otherClasses)
```

Comment:
Why is this a static instead of constructor directly?

Also the signature is ugly, let's just have `__construct(ClassMetadata ...$metadata)` with `assert(count($metadata) !== 0)` if it really MUST be non-empty.